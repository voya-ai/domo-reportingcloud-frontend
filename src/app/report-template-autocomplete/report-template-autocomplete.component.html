<mat-form-field>
   <input
    matInput
    placeholder="Template"
    [matAutocomplete]="autoTemplate"
    [formControl]="templateControl">
</mat-form-field>

<mat-spinner *ngIf="templateLoading"></mat-spinner>

<mat-autocomplete
  #autoTemplate="matAutocomplete"
  [displayWith]="displayTemplate">
  <mat-option
    *ngFor="let option of filteredTemplateOptions"
    [value]="option"
    (onSelectionChange)="selectionChanged(option)">
     {{ option.templateName }}
  </mat-option>
</mat-autocomplete>
