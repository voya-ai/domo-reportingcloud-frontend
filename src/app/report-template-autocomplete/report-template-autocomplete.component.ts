import { Component, Input, Output, EventEmitter, OnInit } from '@angular/core';
import { FormControl } from '@angular/forms';

import { TemplateService } from 'app/template.service';

@Component({
  selector: 'app-report-template-autocomplete',
  templateUrl: './report-template-autocomplete.component.html',
  styleUrls: ['./report-template-autocomplete.component.css']
})
export class ReportTemplateAutocompleteComponent implements OnInit {
  templateControl = new FormControl();
  @Input() templateName: any;
  @Output() onSelectionChange: EventEmitter<any> = new EventEmitter<any>();
  filteredTemplateOptions: string[];
  templateLoading: boolean = false;

  constructor(private _templateService: TemplateService) {}

  ngOnInit() {
    this.templateControl.valueChanges
      .debounceTime(400)
      .distinctUntilChanged()
      .subscribe((val : string) => {
        this.templateLoading = true;
        this._templateService.search(val)
        .toPromise()
        .then(function(result : Array<any>) {
          this.templateLoading = false;
          this.filteredTemplateOptions = result;
        }.bind(this));
      });

    if(this.templateName) {
      this.templateLoading = true;
      this._templateService.get(this.templateName)
        .subscribe((template: any) => {
          this.templateLoading = false;
          this.templateControl.setValue(template);
          this.selectionChanged(template);
        });
    }
  }

  selectionChanged(option: any) {
    this.onSelectionChange.emit(option);
  }

  displayTemplate(template : any) : string {
    return template ? template.templateName : template;
  }
}
