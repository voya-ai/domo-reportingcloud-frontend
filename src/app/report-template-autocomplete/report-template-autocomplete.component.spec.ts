import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { Observable } from 'rxjs/Observable';
import {
  MatAutocompleteModule
} from '@angular/material';

import { TemplateService } from 'app/template.service';

import 'rxjs/add/observable/of';

import { ReportTemplateAutocompleteComponent } from './report-template-autocomplete.component';

class MockTemplateService {}

describe('ReportTemplateAutocompleteComponent', () => {
  let component: ReportTemplateAutocompleteComponent;
  let fixture: ComponentFixture<ReportTemplateAutocompleteComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      providers: [
        { provide: TemplateService, useClass: MockTemplateService }
      ],
      declarations: [ ReportTemplateAutocompleteComponent ],
      imports: [
        MatAutocompleteModule,
        FormsModule,
        ReactiveFormsModule
      ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ReportTemplateAutocompleteComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });
});
