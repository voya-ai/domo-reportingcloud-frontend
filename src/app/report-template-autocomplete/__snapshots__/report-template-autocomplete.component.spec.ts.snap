// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportTemplateAutocompleteComponent should match the snapshot 1`] = `
<app-report-template-autocomplete
  _templateService={[Function MockTemplateService]}
  onSelectionChange={[Function EventEmitter]}
  templateControl={[Function FormControl]}
  templateLoading="false"
>
  <mat-form-field>
    
     
    <input
      aria-autocomplete="list"
      aria-expanded="false"
      aria-owns="mat-autocomplete-1"
      autocomplete="off"
      class="ng-untouched ng-pristine ng-valid"
      matinput=""
      ng-reflect-autocomplete="[object Object]"
      ng-reflect-form="[object Object]"
      placeholder="Template"
      role="combobox"
    />
    
  
  </mat-form-field>
  
  
  
  <mat-autocomplete
    class="mat-autocomplete"
    ng-reflect-display-with="function (template) {
          "
  >
    
  </mat-autocomplete>
  
</app-report-template-autocomplete>
`;
